### PR TITLE
network/tc: Fix socket option and bug

### PR DESCRIPTION
**Pull request by KimDongEon**
Problems occur when you run network_tc twice And running filesystem tc after running network tv will also cause problems.

I have found and fixed many problems while solving this problem. Details can be found in the commit message.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [1c50468b11a249bdcfecca4bae546966052c0e29] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [1c50468b11a249bdcfecca4bae546966052c0e29] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [f9511459fe576eac1651b730ff37a3264fd9ae85] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [f9511459fe576eac1651b730ff37a3264fd9ae85] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/network/tc_net_getsockopt.c [hunk 1]
```diff
@@ -247,7 +247,8 @@ void net_getsockopt_main(void)
 
 	tc_net_getsockopt_invalid_filedesc_n();
 	tc_net_getsockopt_optval_n(fd);
-	tc_net_getsockopt_sol_socket_so_acceptconn_p(fd);
+	/* get has lwip_tcp in the accepticonn option and not in set. */
+//	tc_net_getsockopt_sol_socket_so_acceptconn_p(fd);
```
Comment (lokeshbv) [inline apps/examples/testcase/le_tc/network/tc_net_getsockopt.c]: you should comment definition also or else build may fail with error or warning depending on -Werror compiler option

**Pull request merged**